tests/classes/apiRest/apiRest_test.php

```diff
@@ -0,0 +1,255 @@
+ *
+ * @package    mod
+ * @subpackage accredible
+ * @copyright  Accredible <dev@accredible.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+use mod_accredible\apiRest\apiRest;
+use mod_accredible\client\client;
+
+class mod_accredible_apiRest_testcase extends advanced_testcase {
+    /**
+     * Setup before every test.
+     */
+    public function setUp(): void {
+        $this->resetAfterTest();
+        $this->setAdminUser();
+
+        // Add plugin settings.
+        set_config('accredible_api_key', 'sometestapikey');
+        set_config('is_eu', 0);
+
+        // Unset the devlopment environment variable.
+        $dev_api_endpoint = getenv('ACCREDIBLE_DEV_API_ENDPOINT');
+        putenv('ACCREDIBLE_DEV_API_ENDPOINT');
+
+        $this->mockapi = new class {
+            /**
+             * Returns a mock API response based on the fixture json.
+             * @param string $jsonpath
+             * @return array
+             */
+            public function resdata($jsonpath) {
+                global $CFG;
+                $fixturedir = $CFG->dirroot . '/mod/accredible/tests/fixtures/mockapi/v1/';
+                $filepath = $fixturedir . $jsonpath;
+                return json_decode(file_get_contents($filepath));
+            }
+        };
+    }
+
+    /**
+     * Tests that the api endpoint changes depending on the config.
+     */
+    public function test_api_endpoint() {
+        // When is_eu is NOT enabled.
+        $api = new apiRest();
+        $this->assertEquals($api->api_endpoint, 'https://api.accredible.com/v1/');
+
+        // When is_eu is enabled.
+        set_config('is_eu', 1);
+        $api = new apiRest();
+        $this->assertEquals($api->api_endpoint, 'https://eu.api.accredible.com/v1/');
+
+        // When the environemnt variable is set.
+        putenv('ACCREDIBLE_DEV_API_ENDPOINT=http://host.docker.internal:3000/v1/');
+        $api = new apiRest();
+        $this->assertEquals($api->api_endpoint, 'http://host.docker.internal:3000/v1/');
+    }
+
+    /**
+     * Tests if `GET /v1/credentials/:id` is properly called.
+     */
+    public function test_get_credential() {
+        $mockclient = $this->getMockBuilder('client')
+                           ->setMethods(['get'])
+                           ->getMock();
+
+        // Mock API response data.
+        $resdata = $this->mockapi->resdata('credentials/show_success.json');
+
+        // Expect to call the endpoint once with id.
+        $url = 'https://api.accredible.com/v1/credentials/1';
+        $mockclient->expects($this->once())
+                   ->method('get')
+                   ->with($this->equalTo($url))
+                   ->willReturn($resdata);
+
+        // Expect to return resdata.
+        $api = new apiRest($mockclient);
+        $result = $api->get_credential(1);
+        $this->assertEquals($result, $resdata);
```

Comment:
Good job on writing the unit tests. Just one small comment - you have added tests only for happy paths here. It would be better to cover some basic error scenarios as well.

Reply:
Thanks for the feedback. I added a few more possible error scenarios to the unit tests in apiPhp.php.